src/comparable.cr

```diff
@@ -1,22 +1,40 @@
 # The `Comparable` mixin is used by classes whose objects may be ordered.
 #
 # Including types must provide an `<=>` method, which compares the receiver against
-# another object, returning a negative number, `0`, or a positive number depending
-# on whether the receiver is less than, equal to, or greater than the other object.
+# another object, returning:
+# - a negative number if `self` is less than the other object
+# - a positive number if `self` is greater than the other object
+# - `0` if `self` is equal to the other object
+# - `nil` if `self` and the other object are not comparable
 #
-# `Comparable` uses `<=>` to implement the conventional comparison operators (`<`, `<=`, `==`, `>=`, and `>`).
+# `Comparable` uses `<=>` to implement the conventional comparison operators
+# (`<`, `<=`, `==`, `>=`, and `>`).
```

Comment:
Maybe it could be explicitly mentioned that these operators return `false` when the operands are not comparable (i.e. `<=>` returns `nil`).

Reply:
Done